src/control/lib/hardware/libfabric/bindings.go

```diff
@@ -143,33 +143,19 @@ func (f *fiInfo) hfiUnit() (uint, error) {
 
 // fiGetInfo fetches the list of fi_info structs with the desired provider (if non-empty), or all of
 // them otherwise. It also returns the cleanup function to free the fi_info.
-func fiGetInfo(hdl *dlopen.LibHandle, provider string) ([]*fiInfo, func() error, error) {
-	hint, freeHint, err := fiAllocInfo(hdl)
-	if err != nil {
-		return nil, nil, err
-	}
-	defer func() {
-		if err := freeHint(); err != nil {
-			panic(err)
-		}
-	}()
-
-	if provider != "" {
-		hint.fabric_attr.prov_name = C.CString(provider)
-	}
-
+func fiGetInfo(hdl *dlopen.LibHandle) ([]*fiInfo, func() error, error) {
 	getInfoPtr, err := getLibFuncPtr(hdl, "fi_getinfo")
 	if err != nil {
 		return nil, nil, err
 	}
 
 	var fi *C.struct_fi_info
-	result := C.call_fi_getinfo(getInfoPtr, hint, &fi)
+	result := C.call_fi_getinfo(getInfoPtr, nil, &fi)
```

Comment:
So if I understand the change correctly, the hint was never necessary, even in the case of a non-empty provider string? If this works fine, then I don't have a problem with it, but I'm curious as to whether there's any downside to not providing a hint.

Reply:
If the provider string is non-empty, we need a hint (although apparently the mode needs to be set, which we never did). But we do not, at this point, ever pass a non-empty provider string.

Reply:
Sorry, still not quite following. I think I understand that this change just conforms the code to our current usage (never pass a non-empty provider string, so just removing it from the function signature). Is this likely to change later, though? If not, then I think it's fine to simplify, but if we're just going to have to add it back later then it seems like it'd be better to just handle both cases.